mmv1/compiler.rb

```diff
@@ -139,7 +141,7 @@
 products_for_version = []
 provider = nil
 # rubocop:disable Metrics/BlockLength
-all_product_files.each do |product_name|
```

Comment:
some existing issues doing this.. I think I know why.. trying to see if its fixable

Reply:
looks like its not possible due to `ProviderFileTemplate` having dependency on first generation pass. Probably because it needs to define the resources in the terraform file. hmm

Reply:
should be possible to decouple this dependency. Just need a shallow pass of the product definitions without doing anything too complex